pyspecdata/core.py

```diff
@@ -1333,14 +1334,16 @@ def show_array(x, indent=""):
             else:
                 return x
 
+        if self.data.size < 2:
+            retval = str(self.data.item())
+            if self.get_error() is not None:
+                retval += "±" + str(self.get_error().item())
+            return retval
         retval = show_array(self.data)
-        retval += "\n\t\t+/-"
-        retval += show_array(self.get_error())
-        if (
-            len(self.dimlabels) > 1
-            or len(self.dimlabels) == 0
-            or self.dimlabels[0] != "INDEX"
-        ):
+        if self.get_error() is not None:
+            retval += "\n\t\t±"
+            retval += show_array(self.get_error())
+        if len(self.dimlabels) > 1 or self.dimlabels[0] != "INDEX":
```

Comment:
If the array is just 1D (for example our list of measured voltages) we don't really care/want dimnames attached with the calcualted error. We just want one value in our returned string.

Reply:
more a note to myself but playing around with this and a mwe I see that when I make an nddata with the measured voltages (like in verify.py) and run mean with stderr=True the version before would return the mean with attached error as an nddata array and the error was an array and looked really ugly. But with the updates, the code sees it is 1D and so just attaches a p/m and the calcualted error

Reply:
I think it's useful to know:

- what is the `__str__` method? How is it special?
- the block above involves two changes -- one is to wrap lines 1344 and 1345 with an if statment --> what's that for?  The other is to modify the following if statement --> honestly, that's not very important.

Reply:
(the if statement on line 1346 *is* about dimensionality, in contrast the block that ends on 1341)

Also, for this, note that `"INDEX"` is a "default" dimension name -- if we generate a 1D nddata without explicitly naming the dimension then it calls it `"INDEX"` behind the scenes -- e.g. you get this if you do `a = nddata(r_[0:10])`, although the point of this code is to make sure that you would never explicitly see this unless you asked for `a.dimlabels`

Reply:
here we removed the if the there are no dimlabels (e.g.` len(self.dimlabels)==0`) so only if there is a dimlabel present do we include the array of errors labeled with the dimlabels

Reply:
the first if statement we are saying only if there is an error attached should we display the error as an array as part of the return. (It doesn't make sense to have an empty array attached)

Reply:
the __str__ method I think is so that when we print or want to see the nddata it shows it in the correct layout (e.g. the data points, dimlabels, sizes, and when applicable the error)

Reply:
^this is really helpful for taking the full view of the data and associated dimlabels and axes thereof